router/h2/src/main/scala/io/buoyant/router/h2/ClassifierFilter.scala

```diff
@@ -33,18 +37,30 @@ object ClassifierFilter {
   }
 }
 
-class ClassifierFilter(classifier: ResponseClassifier) extends SimpleFilter[Request, Response] {
+class ClassifierFilter(classifier: H2StreamClassifier) extends SimpleFilter[Request, Response] {
 
   private[this] val successClass = classifier.andThen(_.fractionalSuccess).lift
 
   def apply(req: Request, svc: Service[Request, Response]): Future[Response] = {
-    svc(req).map { rep =>
-      successClass(ReqRep(req, Return(rep))) match {
-        case Some(success) =>
+    svc(req).map {
```

Comment:
https://github.com/linkerd/linkerd/pull/1545 would probably help with this too.  If we can early classify the response, then we should write the response class into a header.  If not, we should wait until the final frame, classify, and write the response class into the final frame if it's a trailers.

The one case that doesn't get covered is if we need the final frame for classification but it's not a Trailers frame.  It's a pretty weird case and I'm okay with punting on it.